src/main/java/net/minecraftforge/event/entity/living/EnderManAngerEvent.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Minecraft Forge - Forge Development LLC
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.event.entity.living;
+
+import net.minecraft.world.entity.monster.EnderMan;
+import net.minecraft.world.entity.player.Player;
+import net.minecraftforge.eventbus.api.Cancelable;
+
+/**
+ * This event is fired on the forge bus before an Enderman detects that a player is looking at them.
+ * It will not be fired if the detection is already prevented by {@link IForgeItem#isEnderMask}
+ * <p>
+ * This event is {@link Cancelable}.
+ * If this event is canceled, the Enderman will not target the player.
+ * <p>
+ * This event does not have a {@link Result}.
+ */
+@Cancelable
+public class EnderManAngerEvent extends LivingEvent {
+    private final Player player;
+
+    public EnderManAngerEvent(EnderMan enderman, Player player)
+    {
+        super(enderman);
```

Comment:
As the event's entity is an `EnderMan`, it will be helpful if this event overrides `getEntity()` and changes the return type to `EnderMan`[^1] (returning the result from the super method using a cast). 

[^1]: This is possible because [return types are covariant](https://www.baeldung.com/java-covariant-return-type) in Java -- an override of a method may have a return type which is a subclass of the overriden method's return type.

Reply:
`getEntity` or `getEntityLiving` or both?

Reply:
I don't think we have a standard, as far as existing source goes it looks like such overrides are not common at all - `AnimalTameEvent`, and `PlayerEvent`, which are both subclasses of `LivingEvent` provide a new method to access their strict types, instead of overriding getEntityLiving.

Arguably `getEntityLiving` should instead be an override on `getEntity` to return a `LivingEntity`
Thus, I would say you should override `getEntity`, which supports potential phasing-out of `getEntityLiving` in the future.

Reply:
Done